src/components/filters/index.js

```diff
@@ -1,3 +1,117 @@
+import React, { useState, useEffect } from 'react';
+import { useRouteMatch, useLocation, useHistory } from 'react-router-dom';
+import { useQueryParams, setQueryParams } from 'url.js';
+import { useYearsPerScenario, useScenariosPerCountry } from 'graphql/queries';
+import { uniqBy, minBy, maxBy, orderBy } from 'lodash';
 import Component from './component';
 
-export default Component;
+const Container = () => {
+  // location, query params
+  const match = useRouteMatch('/:country');
+  const location = useLocation();
+  const history = useHistory();
+  const { country } = (match && match.params) || {};
+  const currentQueryParams = useQueryParams();
+  const { fromY, toYear, scenario } = currentQueryParams;
+
+  // graphql
+  const { data } = useScenariosPerCountry(country);
+  const { data: yearsData } = useYearsPerScenario(country, scenario);
+
+  // state
+  const [years, setYears] = useState(null);
+  const [enabledFromYears, setEnabledFromYears] = useState(null);
+  const [enabledToYears, setEnabledToYears] = useState(null);
+
+  // parsing
+  const scenarios = data && data.scenarios;
+  const parsedScenarios =
+    scenarios &&
+    scenarios.map(sc => ({
+      label: sc.name,
+      value: sc.key
+    }));
+  const parsedYears =
+    years &&
+    years.map(o => ({
+      label: o.year,
+      value: o.year
+    }));
+  const orderedYears = parsedYears && orderBy(parsedYears, 'value');
+
+  // url query params setters
+  const setScenario = sc => {
+    setQueryParams({ scenario: sc }, location, history);
+    const { earliest, latest } = getYearsRange(years);
+    if (earliest && latest && !fromY && !toYear) {
+      setQueryParamsYears(earliest, latest);
+    }
+  }
+  const setFromYear = year =>
+    setQueryParams({ ...currentQueryParams, fromY: year }, location, history);
+  const setToYear = year =>
+    setQueryParams({ ...currentQueryParams, toYear: year }, location, history);
+  const setQueryParamsYears = (fromYear, toY) =>
+    setQueryParams({ ...currentQueryParams, fromY: fromYear, toYear: toY }, location, history);
```

Comment:
Eslint is complaining about the order of those functions:
![image](https://user-images.githubusercontent.com/15097138/67309631-1dc29600-f4f4-11e9-8c3b-185b928e7f63.png)

Reply:
👍 fixed